server/src/tool.clj

```diff
@@ -168,3 +170,29 @@
 (comment
   (start-portal!)
   (tap> {:hello [1 2 3]}))
+
+(def p-lock
```

Comment:
Nit: I would mark this private, and do `defn-` for p-impl, just to keep these all the under the `p` function.